pkg/secrets/cloud/manager_test.go

```diff
@@ -134,6 +136,95 @@ func TestAWSCloudManager_SessionToken(t *testing.T) {
 	t.Setenv("AWS_ACCESS_KEY_ID", creds.AccessKeyID)
 	t.Setenv("AWS_SECRET_ACCESS_KEY", creds.SecretAccessKey)
 	t.Setenv("AWS_SESSION_TOKEN", creds.SessionToken)
+
+	testURL(ctx, t, url)
+}
+
+//nolint:paralleltest // mutates environment variables
+func TestAWSCloudManager_AssumedRole(t *testing.T) {
+	// Regression test for https://github.com/pulumi/pulumi/issues/11482
+	t.Setenv("AWS_REGION", "us-west-2")
+	ctx, cfg, caller := getAwsCaller(t)
+
+	// Make a key with our default config
+	key := createKey(ctx, t, cfg)
+	url := "awskms://" + *key.KeyMetadata.KeyId + "?awssdk=v2"
+
+	// Make a temporary role to assume
+	iamClient := iam.NewFromConfig(cfg)
+	roleName := "test-role-" + randomName(t)
+	assumeRolePolicyDocument := fmt.Sprintf(`{
+		"Version": "2012-10-17",
+		"Statement": {
+			"Effect": "Allow",
+			"Principal": { "AWS": "%s" },
+			"Action": "sts:AssumeRole"
+		}
+	}`, *caller.Arn)
+	role, err := iamClient.CreateRole(ctx, &iam.CreateRoleInput{
+		RoleName:                 &roleName,
+		AssumeRolePolicyDocument: &assumeRolePolicyDocument,
+	})
+	assertNoError(t, err)
```

Comment:
Side-note: Is there a reason we're using assertNoError here instead of require.NoError that comes with testify and has the same behavior?

Reply:
I just didn't think of require.NoError when I wrote these, it would make sense to change to that.